src/data_category.js

```diff
@@ -476,7 +444,7 @@ Blockly.DataCategory.addCreateButton = function (xmlList, workspace, type) {
  * @param {?Array.<string>} opt_secondValue Optional array containing the value
  *     name and shadow type of a second pair of value tags.
  */
-Blockly.DataCategory.addBlock = function (
+function addBlock(
```

Comment:
You could probably get rid of this whole thing and convert the toolbox category to use JSON if you want instead. It's fine to mix dynamic categories that use JSON with static categories that use XML.

Up to you whether you think the time investment is worth the code cleanliness

Reply:
That's a good idea, but I think it can wait for a cleanup pass once we're feature complete. Opened #89 to track.